packages/enzyme-test-suite/test/shared/hooks/useContext.jsx

```diff
@@ -0,0 +1,140 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { expect } from 'chai';
+import wrap from 'mocha-wrap';
+import sinon from 'sinon-sandbox';
+import { Portal } from 'react-is';
+
+import { render } from 'enzyme';
+import getAdapter from 'enzyme/build/getAdapter';
+
+import {
+  describeIf,
+  itIf,
+} from '../../_helpers';
+import {
+  is,
+  REACT16,
+} from '../../_helpers/version';
+
+import {
+  useContext,
+  useState,
+  createContext,
+} from '../../_helpers/react-compat';
+
+export default function describeUseContext({
+  hasHooks,
+  Wrap,
+  WrapRendered,
+  Wrapper,
+  WrapperName,
+  isShallow,
+  isMount,
+  makeDOMElement,
+}) {
+  describeIf(hasHooks, 'hooks: useContext', () => {
+    describe('simple example', () => {
+      let TitleContext;
+      beforeEach(() => {
+        TitleContext = createContext();
+      });
+
+      const UiComponent = props => {
+        let title = useContext(TitleContext);
+        return (
+          <div>
+            <b>UiComponent: </b>
+            {title}
+          </div>
+        );
+      };
+
+      const contextTitle = 'CustomTitle';
+
+      const App = () => {
+        return (
+          <TitleContext.Provider value={contextTitle}>
+            <UiComponent />
+          </TitleContext.Provider>
+        );
+      };
+
+      // TODO; useContext seems to not work in the shallow renderer's initial render
```

Comment:
can we file a bug on react about this?

Reply:
This seems not a bug in react side -- I've tried to write a test to render text with the context initial value and that works in shallow. I think it's more about the bug mentioned in https://github.com/airbnb/enzyme/pull/2008#discussion_r292139433 . But I don't know where the related issue is (e.g. does enzyme has supported `.dive()` into inner component to get the value from outside provider in `shallow`?)

Reply:
yes, ideally that is supported.

Reply:
This seems like a bug. It looks like even we wrap a `Provider` outside `UiComponent`, the context value read by `useContext` still does not change